src/public_key.cc

```diff
@@ -18,7 +18,7 @@ blst::P1 P1::MultiplyBy(
     _point.serialize(out);
     // this should get std::move all the way into the P1 member value
     blst::P1 point{out, public_key_length_uncompressed};
-    point.mult(rand_bytes, rand_bytes_length);
+    point.mult(rand_bytes, rand_bytes_length * 8);
```

Comment:
this is really unintuitive by just looking at the code, why do we need to multiply the length by 8? Why does `rand_bytes_length` not have the right length?

Reply:
It has the length for creating the byte array that gets allocated `uint8_t[rand_bytes_lenght]` and for the random number generator, but the mult functions take number of bits, not number of bytes